### ISSUE TITLE
Document stuck at Loading...

### ISSUE THREAD

**Issue by ErikED** [collective/collective.documentviewer#54]
Hi!

I have been busy trying to install Document Viewer in Plone. All seems to work now, I have been able to add a PDF document and convert it. Now, when I open the page with the PDF file, it keeps saying 'Loading...' forever. (See screenshot right side)
![documentviewer-loading](https://cloud.githubusercontent.com/assets/14904854/10190075/6e0b1d2e-676a-11e5-859d-97ede57ef0f0.jpg)

The only thing I was able to find in the logfile was this:
WARNING Init Class Products.Five.metaclass.Utils has a security declaration for nonexistent method 'convert'

This would indicate possibly that Docsplit is not (correctly) installed? But docsplit is installed and the path to it is included in $PATH. I am also able to connect with putty and use docsplit as a command.

Currently I am really stuck on this. Any help/input would be much appreciated.

Kind regards,
Erik

**Comment by vangheem** [comment]
Sounds like a javascript error.

Can I get a couple things from you:

1) what version of plone
2) what version collective.documentviewer, attempt downgrading to a 3.x version as your problem might be related to the plone 5 compat fixes and upgrade to the doc viewer js.
3) look in the browser console and report on any errors reported there

**Comment by ErikED** [comment]
Thanks for the quick response! :)
1) Plone 4.2.4 (4210)
2) Document Viewer 4.0.0
3) I guess you mean by this: Plone site > Admin > Site Setup > Errors.
No errors are reported here.

Since I am using plone 4.x I did not try downgrading Document Viewer now, since you indicated that problems might have to do with plone 5.

**Comment by gbastien** [comment]
Hi @vangheem, @ErikED 

FYI we recently tried to upgrade too and we had exactly the same problem with Plone 4.3.6, we use now version 3.0.3 but we also pinned z3c.form to 3.2.6 or the configuration of convertable formats is not useable because of a bug in z3c.form 3.2.4 with ordered select widget.

My 2 cents.

Gauthier

**Comment by vangheem** [comment]
@ErikED Nope. I mean the console of the browser. Not the plone site.

![screen shot 2015-09-30 at 8 33 47 am](https://cloud.githubusercontent.com/assets/524413/10194391/01149df6-674e-11e5-981c-65a467053903.png)

**Comment by vangheem** [comment]
Alright, I made a new release that fixes this: https://pypi.python.org/pypi/collective.documentviewer/4.0.2

Seems I had the order of imports mixed for the archetypes template when dealing with the viewer js upgrade. See https://github.com/collective/collective.documentviewer/commit/ad25f9ccc39a9c09ecd785873631fd76112b1126

**Comment by ErikED** [comment]
Hi @vangheem,

Thanks very much for this update!

After installing, I ran into problems however and am getting a proxy error when opening Plone.

This is what I did:
- Plone Site Setup: Deactivate Document Viewer add-on
- Edit buildout.cfg with: collective.documentviewer==4.0.2
- buildout
- plonectl stop + plonectl start

After this i get this error:
![proxy error](https://cloud.githubusercontent.com/assets/14904854/10196238/23d03c44-6792-11e5-9720-80caf2dd7870.jpg)

Downgrading to collective.documentviewer==4.0.0 (and server reboot) fixes this and the plone site pops up again.

**Comment by vangheem** [comment]
Can you look in the event log to see what the error is?

I can not reproduce this. 4.0.2 runs clean for me on 4.3 at least.

**Comment by ErikED** [comment]
I looked at several logfiles. The only thing I could find was in plone_error.log:
[Wed Sep 30 16:46:17 2015] [error] [client 192.168.1.42] (104)Connection reset by peer: proxy: error reading status line from remote server 127.0.0.1:8080
[Wed Sep 30 16:46:17 2015] [error] [client 192.168.1.42] proxy: Error reading from remote server returned by /

Note that as far as I am aware, there is no proxy and the site runs at port 80...

**Comment by vangheem** [comment]
You need to look at the plone log, not the apache logs.

**Comment by ErikED** [comment]
Note that the name of the above log was: PLONE_error.log

Probably you mean one of these:
/usr/local/share/plone/zeocluster/var/client1/event.log:
2015-09-30T16:57:33 INFO ZServer HTTP server started at Wed Sep 30 16:57:33 2015
	Hostname: localhost
	Port: 8080
------
2015-09-30T16:57:33 INFO Zope Set effective user to "plone"
------
2015-09-30T16:57:45 INFO ZEO.ClientStorage zeostorage ClientStorage (pid=3803) created RW/normal for storage: '1'
------
2015-09-30T16:57:45 INFO ZEO.cache created temporary cache file '<fdopen>'
------
2015-09-30T16:57:45 INFO ZEO.ClientStorage zeostorage Testing connection <ManagedClientConnection ('127.0.0.1', 8100)>
------
2015-09-30T16:57:46 INFO ZEO.zrpc.Connection(C) (127.0.0.1:8100) received handshake 'Z3101'
------
2015-09-30T16:57:46 INFO ZEO.ClientStorage zeostorage Server authentication protocol None
------
2015-09-30T16:57:46 INFO ZEO.ClientStorage zeostorage Connected to storage: ('localhost', 8100)
------
2015-09-30T16:57:46 INFO ZEO.ClientStorage zeostorage No verification necessary -- empty cache
------
2015-09-30T16:57:59 INFO ZEO.ClientStorage zeostorage Disconnected from storage: "('localhost', 8100)"

Or 
/usr/local/share/plone/zeocluster/var/zeoserver/zeoserver.log:
2015-09-30T16:55:27 new connection ('127.0.0.1', 49671): <ManagedServerConnection ('127.0.0.1', 49671)>
2015-09-30T16:55:27 new connection ('127.0.0.1', 49672): <ManagedServerConnection ('127.0.0.1', 49672)>
2015-09-30T16:55:27 (127.0.0.1:49672) received handshake 'Z3101'
2015-09-30T16:55:27 new connection ('127.0.0.1', 49673): <ManagedServerConnection ('127.0.0.1', 49673)>
2015-09-30T16:55:27 new connection ('127.0.0.1', 49674): <ManagedServerConnection ('127.0.0.1', 49674)>
2015-09-30T16:55:27 (unconnected) disconnected
2015-09-30T16:55:27 (unconnected) disconnected
2015-09-30T16:55:27 new connection ('127.0.0.1', 49675): <ManagedServerConnection ('127.0.0.1', 49675)>
2015-09-30T16:55:27 new connection ('127.0.0.1', 49676): <ManagedServerConnection ('127.0.0.1', 49676)>
2015-09-30T16:55:27 (127.0.0.1:49676) received handshake 'Z3101'
2015-09-30T16:55:27 (unconnected) disconnected
2015-09-30T16:55:27 (unconnected) disconnected
2015-09-30T16:55:42 (127.0.0.1:49676) disconnected
2015-09-30T16:55:42 (127.0.0.1:49672) disconnected

Hope this helps...

**Comment by vangheem** [comment]
Shows there is no error. Are you sure you just didn't wait long enough for plone to restart?

**Comment by ErikED** [comment]
Yes, I just went back to the site and it is still giving this error.

**Comment by vangheem** [comment]
Can you try running in the foreground?

something like:

plonectl instance fg

Or:

plonectl client1 fg

depending on what your server setup is

**Comment by ErikED** [comment]
Done. I am getting more on screen now. See below.
Currently I have to go away, so I am not sure if I can give any more details today. I will try.

root@plone /# /usr/local/share/plone/zeocluster/bin/plonectl client1 fg
client1: 2015-09-30 17:32:20 INFO ZServer HTTP server started at Wed Sep 30 17:32:20 2015
        Hostname: localhost
        Port: 8080
2015-09-30 17:32:20 INFO Zope Set effective user to "plone"
2015-09-30 17:32:33 INFO ZEO.ClientStorage zeostorage ClientStorage (pid=4979) created RW/normal for storage: '1'
2015-09-30 17:32:34 INFO ZEO.cache created temporary cache file '<fdopen>'
2015-09-30 17:32:34 INFO ZEO.ClientStorage zeostorage Testing connection <ManagedClientConnection ('127.0.0.1', 8100)>
2015-09-30 17:32:34 INFO ZEO.zrpc.Connection(C) (127.0.0.1:8100) received handshake 'Z3101'
2015-09-30 17:32:34 INFO ZEO.ClientStorage zeostorage Server authentication protocol None
2015-09-30 17:32:34 INFO ZEO.ClientStorage zeostorage Connected to storage: ('localhost', 8100)
2015-09-30 17:32:34 INFO ZEO.ClientStorage zeostorage No verification necessary -- empty cache
2015-09-30 17:33:05 INFO ZEO.ClientStorage zeostorage Disconnected from storage: "('localhost', 8100)"
Traceback (most recent call last):
  File "/usr/local/share/plone/zeocluster/parts/client1/bin/interpreter", line 267, in <module>
    execfile(__file__)
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/Startup/run.py", line 76, in <module>
    run()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/Startup/run.py", line 22, in run
    starter.prepare()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/Startup/__init__.py", line 86, in prepare
    self.startZope()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/Startup/__init__.py", line 262, in startZope
    Zope2.startup()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/__init__.py", line 47, in startup
    _startup()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/App/startup.py", line 118, in startup
    load_zcml()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/App/startup.py", line 52, in load_zcml
    load_site()
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/Zope2/App/zcml.py", line 46, in load_site
    _context = xmlconfig.file(site_zcml)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 649, in file
    include(context, name, package)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 207, in feed
    self._parser.Parse(data, isFinal)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 349, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/usr/local/share/plone/buildout-cache/eggs/Zope2-2.13.19-py2.7.egg/OFS/metaconfigure.py", line 46, in loadProducts
    xmlconfig.include(_context, zcml, package=product)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 207, in feed
    self._parser.Parse(data, isFinal)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 349, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/usr/local/share/plone/buildout-cache/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 104, in includePluginsDirective
    includeZCMLGroup(_context, info, filename)
  File "/usr/local/share/plone/buildout-cache/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 30, in includeZCMLGroup
    include(_context, filename, includable_package)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 207, in feed
    self._parser.Parse(data, isFinal)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 349, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 207, in feed
    self._parser.Parse(data, isFinal)
  File "/usr/local/share/plone/Python-2.7/lib/python2.7/xml/sax/expatreader.py", line 349, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 705, in finish
    args = toargs(context, *self.argdata)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 1397, in toargs
    args[str(name)] = field.fromUnicode(s)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/fields.py", line 137, in fromUnicode
    value = self.context.resolve(name)
  File "/usr/local/share/plone/buildout-cache/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 179, in resolve
    mod = __import__(mname, *_import_chickens)
  File "/usr/local/share/plone/buildout-cache/eggs/collective.documentviewer-4.0.2-py2.7.egg/collective/documentviewer/upgrades.py", line 17, in <module>
    from collective.documentviewer.async import queueJob
  File "/usr/local/share/plone/buildout-cache/eggs/collective.documentviewer-4.0.2-py2.7.egg/collective/documentviewer/async.py", line 5, in <module>
    from collective.documentviewer.convert import runConversion
  File "/usr/local/share/plone/buildout-cache/eggs/collective.documentviewer-4.0.2-py2.7.egg/collective/documentviewer/convert.py", line 28, in <module>
    from plone.namedfile.file import NamedBlobImage
zope.configuration.xmlconfig.ZopeXMLConfigurationError: File "/usr/local/share/plone/zeocluster/parts/client1/etc/site.zcml", line 16.2-16.23
    ZopeXMLConfigurationError: File "/usr/local/share/plone/buildout-cache/eggs/Products.CMFPlone-4.2.4-py2.7.egg/Products/CMFPlone/configure.zcml", line 102.4-106.10
    ZopeXMLConfigurationError: File "/usr/local/share/plone/buildout-cache/eggs/collective.documentviewer-4.0.2-py2.7.egg/collective/documentviewer/configure.zcml", line 12.2-12.34
    ZopeXMLConfigurationError: File "/usr/local/share/plone/buildout-cache/eggs/collective.documentviewer-4.0.2-py2.7.egg/collective/documentviewer/upgrades.zcml", line 6.2-12.50
    ImportError: No module named namedfile.file

**Comment by vangheem** [comment]
alright, want to try https://pypi.python.org/pypi/collective.documentviewer/4.0.3 ?

**Comment by ErikED** [comment]
Hi @vangheem,

You are the best! Yesterday evening I did a try with the update and didn't get it running. This morning I had a closer look and somehow plone couldn't write to the logfile anymore (no rights), so it couldn't start. After fixing the file permissions to this manually and doing a restart it is now working!

The Document Viewer looks really great. :) Many thanks for your quick and effective help!

Kind regards,
Erik

**Comment by hodgsoda** [comment]
Hi @vangheem,

Thanks for your support of Document Viewer.  I've been using it for several years and it works very well.  About a year ago I migrated an existing Plone server from Ubuntu to CentOS and installed the Document Viewer prerequisites.  After a few stops and starts I got everything to work, except I am now experiencing the same problem @ErikED was experiencing.  After I add a document (docx, pdf, etc.) and try to view the document, it gets stuck "Loading..."  I tried different browsers and that didn't seem to help.  

I also tried running the server in the foreground (fg) and the log output looks fine; nothing unusual.   I manually verified the files are split up and stored in the file system just fine.
The weird thing is that I also migrated all the _old_ Document Viewer documents from the working Ubuntu server and when I try to view them with Document Viewer on the new server, I do not experience this problem.  The old documents display just fine.  It's just newly added documents that are causing issues for me.  

Note that I opted to use a file folder as the storage type rather than zodb.  I checked the permissions on this folder (/opt/dv771files) and its sub-folders and they look fine. 

Here is my Plone configuration on CentOS 6.9:
Plone 4.3.2 (4307)
CMF 2.2.7
Zope 2.13.21
Python 2.7.6
PIL 1.7.8
Document Viewer 4.0.14 (pinned)

I tried a few different legacy versions of Document Viewer and that didn't seem to help either.  I tried avoiding upgrading Plone, etc. to keep things like-for-like with the Ubuntu server as much as possible.

Any ideas what I'm doing wrong?  Unfortunately I'm not very experienced with CentOS so that doesn't help much.  Thanks in advance.

Best regards,
Dave

**Comment by vangheem** [comment]
Eek, sorry to hear that.

Truth be told, I haven't been testing on plone 4.

If it's stuck on "Loading..." that should mean a js bug. Can you take a look at the browser console and see if there are errors?

**Comment by hodgsoda** [comment]
Hi again.  Thanks for the quick reply.

I checked the browser console using Chrome and Firefox.  There were some errors in the console related to an old theme add-on I was using from Quintagroup, so I removed that egg and rebuilt the site.  That fixed the only error message I was seeing in the console.  So no errors now.

What I find so strange is that I can select the "TEXT" tab in Document Viewer and it correctly shows the text of the document I added.  It's just the "DOCUMENT" tab that get stuck with the "Loading..." message.  So apparently things are working in the back end, it just can't seem to render the document image.

Again, this only happens with new documents that I add to the server.  The old documents from the old server that I migrated over display just fine.  So very strange.

Is there any other diagnostics / logs I can reference?  I agree that this looks like a js issue, but the browser consoles are clean!

Thanks again,
Dave

**Comment by vangheem** [comment]
Can you look at network tab? Can you try changing the image type in the doc viewer settings?

**Comment by hodgsoda** [comment]
Hi Nathan:

I used the network console in Firefox to check the status of a new document  "add" and the retrieval of the same document in the viewer.  The statuses all come back as "200" (OK) so that looks fine.

Then I went into the Document Viewer configuration in the Site Setup and looked at the Image Format.  It was previously set to jpeg so I changed it to gif.  The old legacy documents still retrieve properly, and the files I added with the gif attribute now retrieve properly!  Which is great!  (the files I added with the jpeg property set still don't retrieve, but I only have a handful of them so I can just remove them and re-add).

Thanks so much.  I never would have thought to adjust the Image Format setting!

Dave
